src/vaults/libraries/VaultPricingLib.sol

```diff
@@ -72,9 +83,18 @@ library VaultPricingLib {
         return (_value / 10 ** (PRICE_DECIMALS - decimals)).toUint128();
     }
 
-    /// @dev    Returns the asset decimals and the share decimals for a given vault
-    function getPoolDecimals(address vault) internal view returns (uint8 assetDecimals, uint8 shareDecimals) {
-        assetDecimals = IERC20Metadata(IBaseVault(vault).asset()).decimals();
-        shareDecimals = IERC20Metadata(IBaseVault(vault).share()).decimals();
+    /// @dev    Returns the asset decimals and the share decimals
+    function getPoolDecimals(address shareToken, address asset, uint256 tokenId)
+        internal
+        view
+        returns (uint8 assetDecimals, uint8 shareDecimals)
+    {
+        assetDecimals = getAssetDecimals(asset, tokenId);
+        shareDecimals = IERC20Metadata(shareToken).decimals();
+    }
```

Comment:
Ultra NIT. Is this method really needed? It seems to be just two methods together with no relation. Why not use both directly?